packages/extension/src/serviceWorker/chromeListeners/runtimeOnInstalledListener.ts

```diff
@@ -0,0 +1,133 @@
+ */
+import synchnorousCookieStore from '../../store/synchnorousCookieStore';
+import { getAndParseNetworkCookies } from '../../utils/getAndParseNetworkCookies';
+import attachCDP from '../attachCDP';
+
+export const runtimeOnInstalledListener = async (
+  details: chrome.runtime.InstalledDetails
+) => {
+  synchnorousCookieStore?.clear();
+
+  // @see https://developer.chrome.com/blog/longer-esw-lifetimes#whats_changed
+  // Doing this to keep the service worker alive so that we dont loose any data and introduce any bug.
+  setInterval(() => {
+    chrome.storage.local.get();
+  }, 28000);
+
+  // @todo Send tab data of the active tab only, also if sending only the difference would make it any faster.
+  setInterval(() => {
+    if (Object.keys(synchnorousCookieStore?.tabsData ?? {}).length === 0) {
+      return;
+    }
+
+    Object.keys(synchnorousCookieStore?.tabsData ?? {}).forEach((key) => {
+      synchnorousCookieStore?.sendUpdatedDataToPopupAndDevTools(Number(key));
+    });
+  }, 1200);
+
+  // @todo Send tab data of the active tab only, also if sending only the difference would make it any faster.
+  setInterval(() => {
+    if (Object.keys(synchnorousCookieStore?.tabsData ?? {}).length === 0) {
+      return;
+    }
+
+    Object.keys(synchnorousCookieStore?.tabsData ?? {}).forEach((key) => {
+      getAndParseNetworkCookies(key);
+    });
+  }, 5000);
+
+  if (details.reason === 'install') {
+    await chrome.storage.sync.clear();
+    await chrome.storage.sync.set({
+      allowedNumberOfTabs: 'single',
+      isUsingCDP: false,
+    });
+  }
+
+  if (details.reason === 'update') {
+    await chrome.storage.local.clear();
+    const preSetSettings = await chrome.storage.sync.get();
+    synchnorousCookieStore.tabMode =
+      preSetSettings?.allowedNumberOfTabs ?? 'single';
+    synchnorousCookieStore.globalIsUsingCDP =
+      preSetSettings?.isUsingCDP ?? false;
+
+    if (synchnorousCookieStore.tabMode === 'unlimited') {
+      const allTabs = await chrome.tabs.query({});
+      const targets = await chrome.debugger.getTargets();
+      await Promise.all(
+        allTabs.map(async (tab) => {
+          if (!tab.id || tab.url?.startsWith('chrome://')) {
+            return;
+          }
+
+          synchnorousCookieStore?.addTabData(tab.id);
+
+          if (synchnorousCookieStore.globalIsUsingCDP) {
+            synchnorousCookieStore.initialiseVariablesForNewTab(
+              tab.id.toString()
+            );
+
+            await attachCDP({ tabId: tab.id });
+
+            const currentTab = targets.filter(
+              ({ tabId }) => tabId && tab.id && tabId === tab.id
+            );
+            synchnorousCookieStore?.updateParentChildFrameAssociation(
+              tab.id,
+              currentTab[0].id,
+              '0'
+            );
+          }
+        })
+      );
+    }
+
+    if (
+      preSetSettings?.allowedNumberOfTabs &&
+      Object.keys(preSetSettings).includes('isUsingCDP')
```

Comment:
Optional Chaining can help
```suggestion
      preSetSettings?.isUsingCDP
```

Reply:
Cant do here because it is a boolean variable so we might not know whether or not the variable exists or it is false.